tests/unit/Request.ts

```diff
@@ -0,0 +1,391 @@
+import { Handle, Hash } from 'dojo-core/interfaces';
+import * as lang from 'dojo-core/lang';
+import Promise from 'dojo-core/Promise';
+import request, { providerRegistry, RequestOptions } from 'dojo-core/request';
+import UrlSearchParams from 'dojo-core/UrlSearchParams';
+import registerSuite = require('intern!object');
+import assert = require('intern/chai!assert');
+import * as dstore from 'src/interfaces';
+import Request, { RequestStoreArgs } from 'src/Request';
+import * as mockRequestProvider from './mockRequestProvider';
+
+class Model {
+	constructor(args?: {}) {
+		lang.mixin(this, {
+			_restore: function (Constructor: { new (arg: any): { restored: boolean } }) {
+				// use constructor based restoration
+				const restored = new Constructor(this);
+				restored.restored = true;
+				return restored;
+			},
+			createSubclass: null
+		}, args);
+	}
+
+	_restore(Constructor: { new (arg: any): { restored: boolean } }) {
+		// use constructor based restoration
+		const restored = new Constructor(this);
+		restored.restored = true;
+		return restored;
+	}
+
+	name: string;
+
+	describe() {
+		return 'name is ' + this.name;
+	}
+}
+
+
+class ConcreteRequest<T> extends Request<T> {
+	add(object: any, directives?: dstore.PutDirectives):  any {
+		throw new Error('This Method is abstract');
+	}
+
+	get(id: string | number): Promise<T> | void {
+		throw new Error('This Method is abstract');
+	}
+
+	put(object: T, directives?: dstore.PutDirectives):  Promise<T> {
+		return new Promise(function (resolve) {
+			resolve(object)
+		});
+	}
+
+	remove(id: any): Promise<T | void> {
+		throw new Error('This Method is abstract');
+	}
+}
+
+let store: ConcreteRequest<any>;
+
+const globalHeaders: { [ name: string ]: string } = {
```

Comment:
This is currently used in one place, but I'm not sure it actually has any bearing on the tests.  (The original `Request.js` unit test module also had a `runHeaderTest` function but it was seemingly unused.)